projects/2019_12_16_MultiSig_QT_Wallet.md

```diff
@@ -0,0 +1,90 @@
+# Multi-Sig QT Wallet Project
+* Project Name: MultiSig_QT_Wallet
+* Contact Email: vishnu@gridfinity.com
+* Project participants:
+  * Vishnu Seesahai - Project Owner
+  * *Other participants will be added as needed*
+* Projected duration: 3 months
+* Projected effort: 1 person per month = 3 person-months
+* Requested PKT contribution: 40 million PKT
```

Comment:
Noting that the proposal also commits to continued maintenance.

Reply:
In absence of price discovery, on an open exchange or via. actual utility, we have no way to price this asset at all. If we used a seigniorage estimation, it would be valued closer to 1/ten thousandth of a penny ($.0000012), which is it's min value, without premium. The calc is based on: Price / Coin = Hashes / Block X  Block / Coins X KW / Hash X Price / KW. So, this isn't a land grab by any means, and the commitment to maintenance is assured.

Reply:
I was doing an estimate based on potential market size and max number of PKT on the chain, which of course does not factor in the risk associated with an early stage project, but it does give an idea of what a PKT is valuated at _if_ the network is successful.

Here's one last message that is not inline I hope you can comment on, in case you missed it https://github.com/pkt-cash/ns-projects/pull/7#pullrequestreview-333886785 thanks!

Reply:
Yes, I can certainly respond to this. First, I must point out that it seems to be standard protocol for coin-bearing blockchains to offer a QT wallet, and not having one would seem atypical in the space and likely frowned upon. 

Furthermore, the UX provided by a QT wallet's GUI would be a drastic improvement in experience for non-technical users. This allows us to on-board users who would outright reject the CLI experience wrt. key management and wallet functionality. 

Additionally, PKT Wallet's CLI fails to implement account functionality robustly. Since, PKT Wallet is derived from BTC Wallet, this was obviously incomplete in BTC Wallet as well. I surmise that this is probably related to bitcoin-CLIs deprecation of accounts. However, if you wanted granular control over the sendtoaddress RPC, you need accounts working as expected. Also, sendtoaddress tends to move all transactional change to segwit addresses. This can be difficult to deal with if one executes a lot of transactions, and it totally ruins the balances on the PKT Explorer rich list. All in all, this means that we desire granular control over which addresses we spend from, in other words, we have to execute a raw transaction to make this happen. Obviously, this approach is not expedient, or accessible, for most users. Thus, having this feature available, in a easy-to-use interface, is ideal for user retention. 

Another point to mention is that any installations of the QT wallet expands the number of full nodes on the network, which in turn, improves the consistency of the network. Lastly, the existence of a QT wallet has the potential to propel OTC and block trades through accessibility and simplification of the multi-sig process. Currently, we are using electrum to sign transactions and that wallet is neither native to the PKT Cash chain nor fully integrated with it.

Reply:
These are definitely desirable features, and the NS has experienced usability friction points while signing the last project. @cjdelisle had to craft a Tx to spend from specific (about to be burned) accounts and other NS key holders had to use a custom script to sign blocks:

>granular control over which addresses we spend from

>simplification of the multi-sig process

I also feel okay about the requested PKT amount given the commitment to continued maintenance. This project is definitely beneficial to the PKT project and I am approving this proposal ✅

I'd still like to clarify my earlier question, which is perhaps asking something beyond the scope of your proposed project:

>For example, I have excess bandwidth at home and a Mac laptop. Now there is a user-friendly wallet binary, so I install it... and what's next?

So I now have a QT wallet installed, and many others have done so as well, so the consistency of the network is more secure and clarity around value of PKT. How do you see the path from this point forward to the beginning of a bandwidth market?

Reply:
Yes the goal is to utilize the portability of QT to have this wallet available for multi-platform usage including unix and windows. 

Regarding your second inquiry, it's hard to speculate about that at this time. But, keep in mind that this wallet will be able to run @cjdelisle SPV wallet as well. This means that the wallet supports lnd/lightning, which is a requirement for the bandwidth marketplace. On the roadmap for future versions of the wallet are colored coins, which are also needed for the bandwidth marketplace.  

Furthermore, having a lightweight version of this wallet (SPV), will set up nicely for porting to mobile. One very likely scenario is that the SPV wallet is integrated into Anode's VPN marketplace app. As the VPN marketplace is the first step in creating the bandwidth marketplace, the wallet will be instrumental in this effort.

Reply:
Okay so my understanding here of the expected outcome is:

1. More people run full nodes -> consistent network + more stable PKT pricing
2. Roadmap for wallet -> run on more OSes + SPV on mobile platforms
3. Adoption / integration by first VPN provider -> beginning of a bandwidth marketplace

Of course these are speculative and not part of the scope of this proposal. I am trying to make sure we are all clear on where this project is heading and where this particular project fits into the roadmap.

Thanks for your answers and please comment here if there is any misunderstanding in what I wrote.